examples/jwt-validator/core/src/lib.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use jwt_compact::{
+    alg::{Rsa, RsaPrivateKey, RsaPublicKey},
+    jwk::JsonWebKey,
+    AlgorithmExt, Claims, Header, Token, UntrustedToken,
+};
+use serde::{Deserialize, Serialize};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum TokenError {
+    #[error("Failed to generate token: {0}")]
+    TokenGenerationError(#[from] jwt_compact::CreationError),
+
+    #[error("Failed to validate token: {0}")]
+    TokenValidationError(#[from] jwt_compact::ValidationError),
+
+    #[error("Failed to parse token: {0}")]
+    TokenParseError(#[from] jwt_compact::ParseError),
+}
+
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct CustomClaims {
+    #[serde(rename = "sub")]
+    pub subject: String,
+}
+
+pub struct Issuer {
+    alg: Rsa,
+    s_k: RsaPrivateKey,
+    pub p_k: RsaPublicKey,
+}
+
+impl Issuer {
+    pub fn new(jwk_str: &str) -> Self {
```

Comment:
maybe this should be as a `FromStr` to allow using `parse()`, in case someone wanted to build upon and wanted more graceful error handling or wanted their own custom deserialization of these components?

Reply:
for sure! Was just a suggestion, definitely your call. It wasn't intuitive to me when reading the uses of Issuer/Validator that the constructor was parsing the string.